client/state/data-layer/wpcom/read/site/post-email-subscriptions/update/index.js

```diff
@@ -10,62 +10,65 @@ import { translate } from 'i18n-calypso';
  */
 import { READER_UPDATE_NEW_POST_EMAIL_SUBSCRIPTION } from 'state/action-types';
 import { http } from 'state/data-layer/wpcom-http/actions';
-import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
 import { updateNewPostEmailSubscription } from 'state/reader/follows/actions';
 import { errorNotice } from 'state/notices/actions';
-import getReaderFollowForBlog from 'state/selectors/get-reader-follow-for-blog';
 import { buildBody } from '../utils';
 import { bypassDataLayer } from 'state/data-layer/utils';
+import getReaderFollowForBlog from 'state/selectors/get-reader-follow-for-blog';
 
 import { registerHandlers } from 'state/data-layer/handler-registry';
 
-export function requestUpdatePostEmailSubscription( { dispatch, getState }, action ) {
-	const actionWithRevert = merge( {}, action, {
-		meta: {
-			previousState: get( getReaderFollowForBlog( getState(), action.payload.blogId ), [
-				'delivery_frequency',
-				'email',
-				'post_delivery_frequency',
-			] ),
-		},
-	} );
-	dispatch(
-		http( {
-			method: 'POST',
-			path: `/read/site/${ action.payload.blogId }/post_email_subscriptions/update`,
-			apiVersion: '1.2',
-			body: buildBody( get( action, [ 'payload', 'deliveryFrequency' ] ) ),
-			onSuccess: actionWithRevert,
-			onFailure: actionWithRevert,
-		} )
-	);
+export function requestUpdatePostEmailSubscription( action ) {
+	return ( dispatch, getState ) => {
```

Comment:
Note: I did not move the `previousState` part to the action creator `updateNewPostEmailSubscription ` because it's used in several different places and would break some tests which rely on this action creator returning an object (not a thunk).